src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/SelectResultSet.java

```diff
@@ -366,9 +379,55 @@ private String getFieldName(Field field) {
             }
         }
 
+        if (isSelectAllOnly(query)) {
+            populateAllNestedFields(columns, fields);
+        }
+
         return columns;
     }
 
+    private boolean isFieldSpecifiedExplicitly(Map<String, Field> fieldMap, String fieldName) {
+        return fieldMap.containsKey(fieldName);
+    }
+
+    /**
+     *  Verify if field is property field (object or nested ) and matched by wildcard pattern given in SELECT
+     *  A special case is text field with nested keyword. Ignore it in the case of SELECT * ?
+     */
+    private boolean isFieldPropertyAndMatchWildcard(Map<String, Field> fieldMap, String fieldName) {
+        int lastDot = fieldName.lastIndexOf(".");
+        if (lastDot > -1) {
```

Comment:
np: I think the contract of `lastIndexOf` is it returns -1 if not found, and the index if found, so I would just check that it's not -1, i.e. `lastDot != -1`

Reply:
> np: I think the contract of `lastIndexOf` is it returns -1 if not found, and the index if found, so I would just check that it's not -1, i.e. `lastDot != -1`

Yes, it is. I thought I was using `indexOf`. Will fix it.